server/src/main/java/org/elasticsearch/search/aggregations/InternalAggregations.java

```diff
@@ -60,18 +64,42 @@ public InternalAggregations(List<InternalAggregation> aggregations) {
     }
 
     /**
-     * Reduces the given list of aggregations
+     * Constructs a new aggregation providing its {@link InternalAggregation}s and {@link SiblingPipelineAggregator}s
+     */
+    public InternalAggregations(List<InternalAggregation> aggregations, List<SiblingPipelineAggregator> topLevelPipelineAggregators) {
+        super(aggregations);
```

Comment:
The other ctr could call this one with: `this(aggregations, Collections.emptyList())` ensuring that the `topLevelPipelineAggregators` list is never `null` ?

Reply:
I added to my TODO list to convert this class to Writeable.